src/infrasys/time_series_models.py

```diff
@@ -83,21 +82,36 @@ def length(self) -> int:
         """Return the length of the data."""
         return len(self.data)
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, SingleTimeSeries):
+            raise NotImplementedError
+        is_equal = True
+        for field in self.model_fields_set:
+            if field == "data":
+                if not (self.data == other.data).all():
+                    is_equal = False
+                    break
+            else:
+                if not getattr(self, field) == getattr(other, field):
+                    is_equal = False
+                    break
+        return is_equal
+
     @field_validator("data", mode="before")
     @classmethod
-    def check_data(
-        cls, data
-    ) -> pa.Array | pa.ChunkedArray | pint.Quantity:  # Standarize what object we receive.
+    def check_data(cls, data) -> NDArray | pint.Quantity:  # Standarize what object we receive.
         """Check time series data."""
         if len(data) < 2:
             msg = f"SingleTimeSeries length must be at least 2: {len(data)}"
             raise ValueError(msg)
 
         if isinstance(data, pint.Quantity):
+            if not isinstance(data.magnitude, np.ndarray):
```

Comment:
Please confirm that this is correct. Previously, if the user passed a range object inside a pint Quantity, pint would leave it alone. This guarantees that SingleTimeSeries will always have numpy array.